test/new-e2e/tests/apm/vm_test.go

```diff
@@ -77,26 +94,18 @@ func (s *VMFakeintakeSuite) SetupSuite() {
 	h.MustExecute("sudo groupadd -f -r docker")
 	h.MustExecute("sudo usermod -a -G docker dd-agent")
```

Comment:
We might need to move these into the setup script as well, since group membership changes are not applied to running processes in Linux. It might be flaky if this runs at the right (wrong) time.

Ideally then we won't need to restart the agent at all.